src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -400,4 +419,104 @@ else if (e instanceof PasswordGuardrail.PasswordGuardrailException)
 
         return PasswordObfuscator.obfuscate(e.getMessage());
     }
+
+    private static class JmxFormatter
+    {
+        private static String user(Subject subject)
+        {
+            return String.format("%s", subject == null ? null : subject.getPrincipals().stream().map(Objects::toString).collect(Collectors.joining(", ")));
```

Comment:
I understand this is not a hot path, but following the decision on [this](https://lists.apache.org/thread/65glsjzkmpktzmns6j9wvr4nczvskx36) thread, we may want to stay away from streams in non test code?